central/processbaseline/service/service_impl.go

```diff
@@ -173,9 +205,28 @@ func (s *serviceImpl) DeleteProcessBaselines(ctx context.Context, request *v1.De
 		return response, nil
 	}
 
-	if err := s.dataStore.RemoveProcessBaselinesByIDs(ctx, search.ResultsToIDs(results)); err != nil {
-		return nil, err
+	// go through list of IDs returned from the search results; clear the baseline and remove deployments from observation.
+	for _, r := range results {
+		key, _ := datastore.IDToKey(r.ID)
+
+		// make sure the deployment still exists, if not that process will take care fo the baseline
+		_, exists, err := s.deployments.GetDeployment(ctx, key.GetDeploymentId())
```

Comment:
this check doesn't make sense to me. If the deployment doesn't exist, then we should just delete the baseline right? If it does exists, we should just empty it out, right?

Reply:
Also, when we do the delete, let's do it in a batch delete. Should be faster

Reply:
The idea with that check was if the deployment doesn't exist then there is no point in clearing it or deleting it really.  My assumption being that either the deletion of a deployment is in progress and the baseline will be deleted as part of that process or the pruning will catch up to it the next time it runs garbage collection.  But while here, probably simple enough to go ahead and delete it.